src/parser/parse_heredoc.c

```diff
@@ -0,0 +1,37 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        ::::::::            */
+/*   parse_heredoc.c                                    :+:    :+:            */
+/*                                                     +:+                    */
+/*   By: lvan-bus <lvan-bus@student.codam.n>          +#+                     */
+/*                                                   +#+                      */
+/*   Created: 2023/05/05 09:27:00 by lvan-bus      #+#    #+#                 */
+/*   Updated: 2023/05/05 09:27:02 by lvan-bus      ########   odam.nl         */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include <minishell.h>
+
+int32_t	here_doc(char *delimiter, int fd_write_end)
+{
+	char	*line;
+
+	if (!delimiter || ft_strlen(delimiter) == 0)
+		return (ERROR);
+	while (1)
+	{
+		signal(SIGINT, signal_ctrl_c_heredoc);
```

Comment:
this one can happen outside of the while loop right? Or should it be reset again after the readline has finished?

Reply:
you're right, done